### PR TITLE
Don't push all tags when sha256 builds just one

### PR DESCRIPTION
**Pull request by michaelfig**
I don't yet know how to describe the problem, except for that sha256 sends thousands of pushes if you use it with an image that doesn't have its own tag.

Forcing the tag to `:skaffold` (`:latest` would work, too) makes it behave properly.

### CODE REVIEW

File: pkg/skaffold/build/tag/sha256.go [hunk 1]
```diff
@@ -31,5 +36,17 @@ func (c *ChecksumTagger) Labels() map[string]string {
...
+	}
+	noRepo := repoPortRe.ReplaceAllLiteralString(imageName, "")
+
+	if strings.Contains(noRepo, ":") {
+		// They have a tag, so use it.
+		return imageName, nil
+	}
+
+	// No supplied tag, so use skaffold.
+	return imageName + ":skaffold", nil
```
Comment (dgageot) [inline pkg/skaffold/build/tag/sha256.go]: I'd rather we use `:latest`, here

File: pkg/skaffold/build/tag/sha256.go [hunk 2]
```diff
@@ -31,5 +36,17 @@ func (c *ChecksumTagger) Labels() map[string]string {
 }
 
 func (c *ChecksumTagger) GenerateFullyQualifiedImageName(workingDir, imageName string) (string, error) {
-	return imageName, nil
+	// Strip repositories that have a port number.
+	if repoPortRe == nil {
```
Comment (dgageot) [inline pkg/skaffold/build/tag/sha256.go]: Please, use something like `parsed, err := docker.ParseReference(tag)` to parse image names.
Also, image names on artifacts should never have a tag. We should maybe enforce that somewhere else.

**Comment by dgageot** [comment]
@michaelfig Could you create a ticket with a procedure to reproduce? I couldn't reproduce the issue here.

File: pkg/skaffold/build/tag/sha256.go (hunk 2 above)
Comment (michaelfig) [reply pkg/skaffold/build/tag/sha256.go]: #1623 is where I explained a use case for having tags on built/pushed artifacts.

Thanks so much for your feedback!

**Comment by michaelfig** [comment]
@dgageot working on finding a simple example for a new issue.

**Comment by michaelfig** [comment]
@dgageot #1638 describes this problem in detail, with minimal steps to reproduce.

On my system, I have a lot of tags on one image, so it's especially slow and onerous to push them all.

**Pull request merged**